### PR TITLE
feat(geo): Add name setter and move constructor to `TrackingVolume`

### PR DESCRIPTION
**Pull request by paulgessinger**

### CODE REVIEW

File: Core/include/Acts/Geometry/TrackingVolume.hpp [hunk 1]
```diff
@@ -115,6 +115,8 @@ class TrackingVolume : public Volume {
   TrackingVolume(const TrackingVolume&) = delete;
   TrackingVolume& operator=(const TrackingVolume&) = delete;
 
+  TrackingVolume(TrackingVolume&&) = default;
```
Comment (andiwand) [inline Core/include/Acts/Geometry/TrackingVolume.hpp:118]:
```suggestion
  TrackingVolume(TrackingVolume&&) = default;
  TrackingVolume& operator=(TrackingVolume&&) = default;
```

do we also want a move assign?

**Review by andiwand — APPROVED** [review]

File: Core/include/Acts/Geometry/TrackingVolume.hpp (hunk 1 above)
Comment (paulgessinger) [reply Core/include/Acts/Geometry/TrackingVolume.hpp:118]: I didn't need one, but can add it.

**Comment by paulgessinger** [comment]
@andiwand added now.

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [c7ee59f2021db2c42ac363965516b25e24dadec0]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42711242) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/7993146)!

**Pull request merged**